src/runtime_view_proxy.js

```diff
@@ -0,0 +1,187 @@
+/**
+ * @license
+ * Copyright 2023 The Emscripten Authors
+ * SPDX-License-Identifier: MIT
+ */
+
+// Chrome does not allow TypedArrays with more than 4294967296 elements
+// We'll create proxy objects for HEAP(U)8 when memory is > 4gb and for HEAP(U)16 when > 8gb
+// https://bugs.chromium.org/p/v8/issues/detail?id=4153
+var maxArraySize = Math.min(b.byteLength, 4 * 1024 * 1024 * 1024 - 8);
+/**
+ * @param {string} type - Heap type
+ * @param {number} [offset] - Heap offset
+ * @param {number} [length] - typed array length
+ */
+function getHeapBlock(type, offset, length) {
+  if (!offset) {
+    offset = 0
+  }
+
+  const heap = wasmMemory.buffer
+
+  // we should always limit the length to maxArraySize
+  function createTypedArray (arrayType, offset, length) {
+    const bpe = arrayType.BYTES_PER_ELEMENT;
+    return new arrayType(heap, offset, length || Math.min(heap.byteLength - offset * bpe, maxArraySize / bpe));
+  }
+
+  switch (type) {
+    case 'i1':
+    case 'i8':
+      return createTypedArray(Int8Array, offset, length);
+    case 'u1':
+    case 'u8':
+      return createTypedArray(Uint8Array, offset, length);
+    case 'i16':
+      return createTypedArray(Int16Array, offset, length);
+    case 'u16':
+      return createTypedArray(Uint16Array, offset, length);
+    case 'i32':
+      return createTypedArray(Int32Array, offset, length);
+    case 'u32':
+      return createTypedArray(Uint32Array, offset, length);
+    case 'f32':
+      return createTypedArray(Float32Array, offset, length);
+    case 'f64':
+      return createTypedArray(Float64Array, offset, length);
+    case 'i64':
+      return createTypedArray(BigInt64Array, offset, length);
+    case '*':
+    case 'u64':
+      return createTypedArray(BigUint64Array, offset, length);
+    default:
+      throw new Error('Invalid type');
+  }
+}
+
+function createProxyHandler (type, heapBlocks) {
+  const firstHeapBlock = heapBlocks[0]
```

Comment:
For this type of code that gets embedded in the output we are still using `let` instead of `const`/`var`.  Would you mind converting all these `const` to `let` in this file?